pkg/terraform/terraform.go

```diff
@@ -115,54 +115,62 @@ func (t *Terraform) socketPath(c interfaces.Cluster) string {
 }
 
 func (t *Terraform) Prepare(cluster interfaces.Cluster) error {
-	select {
-	case <-t.ctx.Done():
-		return t.ctx.Err()
-	default:
+	if err := t.checkDone(); err != nil {
+		return err
 	}
 
 	// generate tf code
 	if err := t.GenerateCode(cluster); err != nil {
 		return fmt.Errorf("failed to generate code: %s", err)
 	}
 
-	select {
-	case <-t.ctx.Done():
-		return t.ctx.Err()
-	default:
+	if err := t.checkDone(); err != nil {
+		return err
 	}
 
 	// symlink tarmak plugins into folder
 	if err := t.preparePlugins(cluster); err != nil {
 		return fmt.Errorf("failed to prepare plugins: %s", err)
 	}
 
-	select {
-	case <-t.ctx.Done():
-		return t.ctx.Err()
-	default:
+	if err := t.checkDone(); err != nil {
+		return err
 	}
 
-	// run init
-	if err := t.command(
-		cluster,
-		[]string{
+	type setupCommand struct {
+		log, name string
+		args      []string
+	}
+
+	for _, c := range []setupCommand{
+		{log: "initialising terraform", name: "init", args: []string{
 			"terraform",
 			"init",
 			"-get-plugins=false",
 			"-input=false",
 		},
-		nil,
-		nil,
-		nil,
-	); err != nil {
-		return fmt.Errorf("failed to run terraform init: %s", err)
-	}
+		},
+		{log: "formatting terraform code", name: "fmt", args: []string{
+			"terraform",
+			"fmt",
+			"-list=false",
+		},
+		},
+		{log: "validating terraform code", name: "validate", args: []string{
+			"terraform",
+			"validate",
+		},
+		},
+	} {
```

Comment:
What is the benefit of running that every time we plan/apply and destroy.

We should make sure the generated code is always formatted and valid. This should be ensured at PR time through a go test instead (IMHO)

(See my feedback in the review)